File: k.py
import aoc, e
from collections import defaultdict as dd
s = aoc.sread('i.11', int, ',')
def do(grid, f):
    r = e.runtime(s)
    pos = [0,0]
    direct = 0
    while 1:
        try:
            po = tuple(pos) #lists cannot be hashed
            v = grid.get(po, 0)
            grid[po] = r.send(v)
            turn = r.next()
            r.next()
            #speedcode for turning & moving
            if turn: direct += 1
            else: direct -= 1
            if direct < 0: direct += 4
            if direct > 3: direct -= 4
            pos[direct%2==0] += [-1,1][direct<2]
        except Exception:
            print f(grid)
            return
do(dd(lambda: 0), len)
do(dd(lambda: 0, {(0,0):1}), lambda g: aoc.toGrid(g, {0:'.', 1:'#'}))

File: js/k.js
const aoc = require('./aoc');
const s = aoc.sread('i.11', Number, ',')
function paint(grid, f) {
    const r = require('./e').runtime(s);
    let pos = [0,0];
    let dir = 0;
    while (1) {
        grid.set(aoc.toHashable(pos.slice(0)), r.next(grid.get(aoc.toHashable(pos))).value);
        if (r.next().value) dir += 1;
        else    dir -= 1;
        if (log = r.next().done) break;
        if (dir > 3) dir -= 4;
        else if (dir < 0) dir += 4;
        pos[Number(dir%2==0)] += [-1,1][Number(dir < 2)];
    }
    console.log(f(grid));
}
paint(new aoc.defaultMap(()=>0), d => d.size);
paint(new aoc.defaultMap(()=>0).set(aoc.toHashable([0,0]),1), g => aoc.toGrid(g, {0:'.', 1:'#'}))
